### PR TITLE
AOM-47: Fix search description bug

### PR DESCRIPTION
**Pull request by tduyile04**
## JIRA TICKET NAME:

AOM-47: [Fix search description bug](https://issues.openmrs.org/browse/AOM-47)

### SUMMARY:

This PR aims to fix the bug on the search feature and extends its functionality to work for `OWA` and modules.

### CODE REVIEW

File: app/js/components/common/Header.jsx [hunk 1]
```diff
@@ -55,15 +55,15 @@ export default class Header extends Component {
     this.fetchLocation('/session').then((response) => {
       this.setState((prevState, props) => {
         return {
-          currentUser: response.user.display
+          currentUser: response.user && response.user.display
```
Comment (andela-dbamidele) [inline app/js/components/common/Header.jsx]: I love the fact that you are checking if the `user` value is not null before setting the state. But I think you should consider if `response.user` is `null` as well except if you don't want the state to change if it's null. `response.user ? response.user.display : ''` should do the job for you.

Nice work though 👍

**Review by andela-dbamidele — COMMENTED** [review]
Great job 👍

File: app/js/components/common/Header.jsx (hunk 1 above)
Comment (tduyile04) [reply app/js/components/common/Header.jsx]: Thanks @andela-dbamidele for the nitpick. Would put this into effect.

File: app/js/components/manageApps/AddonList.jsx [hunk 2]
```diff
@@ -9,40 +9,41 @@
...
       {
         appList.map((app, key) => {
           return (
             <tr key={key}>
-              <td onClick={() => openPage(app)}>
+              <td onClick={() => openPage(app.appDetails)}>
                 <img
-                  src={app.icons !== null && app.icons !== undefined ? 
-                    `/${location.href.split('/')[3]}/owa/openmrs-addonmanager/${app.icons[48]}` : 
+                  src={app.appDetails && app.appDetails.icon !== null && app.appDetails.icons !== undefined ?
```
Comment (Ewanjiru) [inline app/js/components/manageApps/AddonList.jsx]: Your line seem too long more than 80 chars

**Comment by brayoh** [comment]
Looks great 👍

File: app/js/components/manageApps/ManageApps.jsx [hunk 3]
```diff
@@ -85,11 +85,16 @@ export default class ManageApps extends React.Component {
   }
 
   handleApplist() {
+    const resultData = [];
     this.apiHelper.get('/owa/applist').then(response => {
+      resultData.push({
+        appDetails: response[0],
+        install: false
```
Comment (jacksono) [inline app/js/components/manageApps/ManageApps.jsx]: You are declaring 'resultData with 'const' yet you are changing its value, it  looks like you should be use 'let' instead. Check out this similar case https://stackoverflow.com/questions/40825010/const-or-let-in-react-component
There could other similar in your code, please check through.

File: app/js/components/manageApps/ManageApps.jsx [hunk 4]
```diff
@@ -360,73 +371,88 @@ export default class ManageApps extends React.Component {
...
+    if (addOnFound.length > 0) {
+      resultData.push({
+        appDetails: addOnFound,
+        downloadUri: null,        
+        install: false
+      });
+    }
+    if(resultData,length > 0) {
+      this.setState((prevState, props) => {
+        return {
```
Comment (jacksono) [inline app/js/components/manageApps/ManageApps.jsx]: Is 'resultData,length' really what you meant to write or 'resultData.length'

File: app/js/components/manageApps/ManageApps.jsx (hunk 3 above)
Comment (andela-dbamidele) [reply app/js/components/manageApps/ManageApps.jsx]: I think it's right to use `const` in this context. `resultData` is not totally been reassigned, it's just an addition of values. And by the way, `eslint` will flag it if you use `let` 🙂

Comment (tduyile04) [reply app/js/components/manageApps/ManageApps.jsx]: Thanks @jacksono @andela-dbamidele Following up with the thread, I only mutated the data in `resultData` and didn't re-assign it. `const` only prevents re-assignment, so it pretty much works for this use case.

File: app/js/components/manageApps/ManageApps.jsx (hunk 4 above)
Comment (tduyile04) [reply app/js/components/manageApps/ManageApps.jsx]: Wow, thanks a lot. @jacksono would correct this `typo` asap

File: app/js/components/manageApps/AddonList.jsx (hunk 2 above)
Comment (tduyile04) [reply app/js/components/manageApps/AddonList.jsx]: Thanks @Ewanjiru  would refactor that 👍

File: app/js/components/manageApps/ManageApps.jsx [hunk 5]
```diff
@@ -31,13 +31,12 @@ export default class ManageApps extends React.Component {
...
       selectedApp: null,
-      searchResults: [],
-      install: false,
       downloadUri: null,
       deleteStatus: false,
       addonAlreadyInstalled: null,
       files: null,
-      displayInvalidZip: false
+      displayInvalidZip: false,
+      result: []
```
Comment (mkiterian) [inline app/js/components/manageApps/ManageApps.jsx]: What does `result` store? Where is it used?

File: app/js/components/manageApps/AddAddon.jsx [hunk 6]
```diff
@@ -16,12 +16,13 @@ export default class AddAddon extends Component {
...
       disableClick: true,
       dropzoneRef: {},
-    }
+    };
     this.showMessage = this.showMessage.bind(this);
   }
 
   showMessage() {
-    if (this.props.files == null || this.props.files.length < 1) {
+    const { files } = this.props;
```
Comment (LWanjiru) [inline app/js/components/manageApps/AddAddon.jsx:24]: Awesome refactoring here. The function looks cleaner.

Comment (tduyile04) [reply app/js/components/manageApps/AddAddon.jsx:24]: Thanks @LWanjiru

File: app/js/components/manageApps/AddonList.jsx [hunk 7]
```diff
@@ -9,40 +9,48 @@
...
+                    `/${location.href.split('/')[3]}/owa/openmrs-addonmanager/${app.appDetails.icons[48]}` : 
                     `./img/omrs-button.png`}
                   alt="addon logo"
                 />
               </td>
-              <td onClick={() => openPage(app)}>
-                <div className="addon-name">{app.name}</div>
-                <div><h5  className="addon-description">{app.description}</h5></div>
+              <td onClick={() => openPage(app.appDetails)}>
+                <div className="addon-name">{app.appDetails ? app.appDetails.name : ''}</div>
```
Comment (LWanjiru) [inline app/js/components/manageApps/AddonList.jsx]: Curious question: I see here that you return either a name or an empty string dependent on the evaluation of app.appDetails. Why did you opt to return an empty string?

Comment (tduyile04) [reply app/js/components/manageApps/AddonList.jsx]: Hey @LWanjiru It was a quick fix to resolve a bug, working on the update. Thanks for spotting it out.

File: app/js/components/manageApps/ManageApps.jsx (hunk 5 above)
Comment (tduyile04) [reply app/js/components/manageApps/ManageApps.jsx]: @mkiterian Thanks for noting that, forgot to remove it.

**Comment by mkiterian** [comment]
Looks good. Great job!

**Pull request merged**